File: src/types/plan.ts
export type PhaseStatus = 'pending' | 'in_progress' | 'done'

export interface Phase {
  id: string
  planId: string
  name: string
  description?: string
  filePath?: string
  order: number
  orderIndex?: number
  status: PhaseStatus
  createdAt?: string
  updatedAt?: string
}

export interface Plan {
  id: string
  deckId: string
  name: string
  reportPath?: string
  planPath?: string
  phases: Phase[]
  createdAt: string
  updatedAt?: string
}

File: src-tauri/src/models/plan.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum PhaseStatus {
    Pending,
    InProgress,
    Done,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase")]
pub struct Phase {
    pub id: String,
    pub plan_id: String,
    pub name: String,
    pub description: Option<String>,
    pub file_path: Option<String>,
    pub order_index: i32,
    pub status: PhaseStatus,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase")]
pub struct Plan {
    pub id: String,
    pub deck_id: String,
    pub name: String,
    pub report_path: Option<String>,
    pub plan_path: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
